perimetre_asa/asaperimetre.py

```diff
@@ -49,7 +51,24 @@ def initGui(self):
         """Init the user interface."""
         self.initProcessing()
         self.dock = AsaPerimetreDock()
-        iface.addDockWidget(Qt.RightDockWidgetArea, self.dock)
+        self.iface.addDockWidget(Qt.RightDockWidgetArea, self.dock)
+
+        action = self.dock.toggleViewAction()
+
+        self.menu = QMenu("&Périmètre d'ASA")
```

Comment:
1. All variables created in `self` should be created in the `init` function. This is also raising a warning in a Python IDE.

2. As UX improvement, better to avoid the use of a menu when there is a single item. Like opening the Python console from the "plugin" menu, there isn't a menu going to the right. It's better design.

Reply:
The point number 2 is still here, no ?
Just add your action in the plugin menu, no need to create a QMenu.

Also when there is GUI stuff loaded by QGIS, you should also check to unload them in the `unload` function below.
https://github.com/3liz/qgis-pgmetadata-plugin/blob/master/pg_metadata/pg_metadata.py#L97